Common/Packets/Controls.cs

```diff
@@ -96,6 +96,21 @@ public class Controls
         [JsonProperty(PropertyName = "oTrainingLimits")]
         public LeakyBucketControlParameters TrainingLimits;
 
+        /// <summary>
+        /// Limits the total size of storage used by <see cref="IObjectStore"/>
+        /// </summary>
+        [JsonProperty(PropertyName = "storageLimitMB")]
+        public int StorageLimitMB;
+
+        /// <summary>
+        /// Limits the number of files to be held under the <see cref="IObjectStore"/>
+        /// </summary>
+        [JsonProperty(PropertyName = "storageFileCountMB")]
+        public int StorageFileCount;
+
+        [JsonProperty(PropertyName = "persistenceIntervalSeconds")]
```

Comment:
Please add a comment;
"Delay between persisting data to the store"

Reply:
What are your thoughts on `IObjectStore` sporting some events? This enables inverting the dependencies so we're not injecting dependencies into the `IObjectStore`, instead, `QCAlgorithm` would bind handlers to the `IObjectStore.SaveFailed` event (or similar, perhaps a more generic `IObjectStore.ErrorRaised` event would be better).

Wiring up dependencies using events keeps components loosely coupled and also precludes us from having to add the _next_ dependency (whatever that may be). All that being said, if you prefer to push the dependencies into the `IObjectStore`, I'm happy to oblige :)